### PR TITLE
DESENG-611: Splitting translations based on language and common

### PR DESCRIPTION
**Pull request by ratheesh-aot**
Issue #: https://apps.itsm.gov.bc.ca/jira/browse/DESENG-611

*Description of changes:*
- Removed Tenant based translations from language files
  - Set TODO statements to replace tenant-related values from the backend. [Ticket for this](https://apps.itsm.gov.bc.ca/jira/browse/DESENG-612)
  - Splitting translations based on language files and adding a common translation file for items common across all languages

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

**Comment by ratheesh-aot** [comment]
> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=bcgov_met-public&pullRequest=2525) **Quality Gate failed**
> Failed conditions  [B Maintainability Rating on New Code](https://sonarcloud.io/dashboard?id=bcgov_met-public&pullRequest=2525) (required ≥ A)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=bcgov_met-public&pullRequest=2525)
> 
>  Catch issues before they fail your Quality Gate with our IDE extension  [SonarLint](https://www.sonarsource.com/products/sonarlint/features/connected-mode/?referrer=pull-request)

Its because of the TODO Statements to replace tenant specific values from database which will done in another story

File: met-web/src/hooks.ts [hunk 1]
```diff
@@ -8,15 +8,26 @@ export const useAppDispatch = () => useDispatch<AppDispatch>();
 export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
 
 export const useAppTranslation = () => {
-    const translate = useTranslation();
-    const tenantId = sessionStorage.getItem('tenantId');
+    // Every language has its own default and common namespaces
+    const translate = useTranslation(['default', 'common']);
 
     const { t } = translate;
```
Comment (Baelx) [inline met-web/src/hooks.ts:15]: The language system in MET is inherently a bit confusing. The comments in this file are great so far, and maybe we could add a higher-level one above this function. Would this comment describe the intended behaviour?

```suggestion
/**
* Look to see if a string has a tenant-level translation (if it's related to an engagement, survey, widget, static       
* tenant text, etc.). If none is found, look for a common static text translation found in the common locale file.
**/
```

**Review by Baelx — APPROVED** [review]
Thanks! You're free to merge but it'd be appreciated if you could maybe change variable names and add a bit more commenting so it can be clear how the complex language system of the app works.

File: met-web/src/App.tsx [hunk 2]
```diff
@@ -137,17 +138,17 @@ const App = () => {
 
     const getTranslationFile = async (languageId: string) => {
         try {
-            const translationFile = await import(`./locales/${languageId}/${tenant.id}.json`);
+            const translationFile = await import(`./locales/${languageId}.json`);
```
[the remark this answers is not in the document]
Comment (ratheesh-aot) [reply met-web/src/App.tsx]: Changed to **localeId**

File: met-web/src/App.tsx [hunk 3]
```diff
@@ -115,9 +115,10 @@ const App = () => {
         if (!tenant.id) {
             return;
         }
-
         try {
-            const supportedLanguages = Object.values(Languages);
+            const supportedLanguages: string[] = Object.values(Languages);
+            // Adding keyword `common` into supportedLanguages to fetch locale/common.json
+            supportedLanguages.push('common');
```
[the remark this answers is not in the document]
Comment (ratheesh-aot) [reply met-web/src/App.tsx]: Separated the way to load common translation file

File: met-web/src/hooks.ts (hunk 1 above)
Comment (ratheesh-aot) [reply met-web/src/hooks.ts]: Updated the comments

**Review by Baelx — APPROVED** [review]
Thanks for this!

**Pull request merged**